engine/Sandbox.Engine/Scene/Components/Game/Prop.cs

```diff
@@ -546,6 +565,69 @@ public List<Gib> CreateGibs()
 		return gibs;
 	}
 
+	/// <summary>
+	/// Gets the effective impact damage value. If ImpactDamage is not set,
+	/// calculates a default value based on the prop's mass.
+	/// </summary>
+	private float GetEffectiveImpactDamage()
+	{
+		if ( ImpactDamage > 0 )
+			return ImpactDamage;
+
+		// Calculate from mass if not explicitly set
+		var rb = Components.Get<Rigidbody>();
+		return rb.IsValid() && rb.PhysicsBody.IsValid() ? rb.PhysicsBody.Mass / 10f : 10f;
+	}
+
+	/// <summary>
+	/// Called when this prop starts colliding with another object.
+	/// Handles velocity-based damage for high-speed impacts.
+	/// </summary>
+	void ICollisionListener.OnCollisionStart( Collision collision )
+	{
+		if ( !EnableImpactDamage ) return;
+		if ( IsProxy ) return;
+
+		var speed = collision.Contact.Speed.Length;
+		var minSpeed = MinImpactDamageSpeed;
+		if ( minSpeed <= 0 )
+			minSpeed = 500f;
+
+		if ( speed <= minSpeed )
+			return;
+
+		var impactDmg = GetEffectiveImpactDamage();
+		var damageMultiplier = speed / minSpeed;
+
+		// This prop takes damage from high speed impacts
+		if ( Health > 0 )
+		{
+			var selfDamage = damageMultiplier * impactDmg;
+			var damageInfo = new DamageInfo( selfDamage, collision.Other.GameObject, collision.Other.GameObject )
+			{
+				Position = collision.Contact.Point
+			};
+			damageInfo.Tags.Add( "physics_impact" );
+
+			OnDamage( damageInfo );
+		}
+
+		// The other object takes more damage
+		var otherDamageable = collision.Other.GameObject?.GetComponentInParent<IDamageable>();
```

Comment:
If the other object is a prop, will OnCollisionStart get called for that one too, resulting in double damage?